webapp/channels/src/components/window_size_observer/WindowSizeObserver.tsx

```diff
@@ -0,0 +1,93 @@
+            break;
+        case mediumSidebarMediaQuery.matches:
+            dispatch(setLhsSize(SidebarSize.MEDIUM));
+            dispatch(setRhsSize(SidebarSize.MEDIUM));
+            break;
+        case smallSidebarMediaQuery.matches:
+            dispatch(setLhsSize(SidebarSize.SMALL));
+            dispatch(setRhsSize(SidebarSize.SMALL));
+            break;
+        }
+    }, [dispatch]);
+
+    const setSidebarSizeWhenWindowResized = useCallback(throttle(() => {
+        dispatch(setLhsSize());
+        dispatch(setRhsSize());
+    }, 100), []);
+
+    const handleSidebarMediaQueryChangeEvent = useCallback((e: MediaQueryListEvent) => {
+        if (e.matches) {
+            updateSidebarSize();
+        }
+    }, [updateSidebarSize]);
+
+    useEffect(() => {
+        updateSidebarSize();
+
+        if (smallSidebarMediaQuery.addEventListener) {
+            xLargeSidebarMediaQuery.addEventListener('change', handleSidebarMediaQueryChangeEvent);
+            largeSidebarMediaQuery.addEventListener('change', handleSidebarMediaQueryChangeEvent);
+            mediumSidebarMediaQuery.addEventListener('change', handleSidebarMediaQueryChangeEvent);
+            smallSidebarMediaQuery.addEventListener('change', handleSidebarMediaQueryChangeEvent);
+        } else if (smallSidebarMediaQuery.addListener) {
+            xLargeSidebarMediaQuery.addListener(handleSidebarMediaQueryChangeEvent);
+            largeSidebarMediaQuery.addListener(handleSidebarMediaQueryChangeEvent);
+            mediumSidebarMediaQuery.addListener(handleSidebarMediaQueryChangeEvent);
+            smallSidebarMediaQuery.addListener(handleSidebarMediaQueryChangeEvent);
+        } else {
+            window.addEventListener('resize', setSidebarSizeWhenWindowResized);
+        }
+
+        return () => {
+            if (smallSidebarMediaQuery.removeEventListener) {
+                xLargeSidebarMediaQuery.removeEventListener('change', handleSidebarMediaQueryChangeEvent);
+                largeSidebarMediaQuery.removeEventListener('change', handleSidebarMediaQueryChangeEvent);
+                mediumSidebarMediaQuery.removeEventListener('change', handleSidebarMediaQueryChangeEvent);
+                smallSidebarMediaQuery.removeEventListener('change', handleSidebarMediaQueryChangeEvent);
+            } else if (smallSidebarMediaQuery.removeListener) {
+                xLargeSidebarMediaQuery.removeListener(handleSidebarMediaQueryChangeEvent);
+                largeSidebarMediaQuery.removeListener(handleSidebarMediaQueryChangeEvent);
+                mediumSidebarMediaQuery.removeListener(handleSidebarMediaQueryChangeEvent);
+                smallSidebarMediaQuery.removeListener(handleSidebarMediaQueryChangeEvent);
+            } else {
+                window.removeEventListener('resize', setSidebarSizeWhenWindowResized);
+            }
+        };
+    }, [handleSidebarMediaQueryChangeEvent, setSidebarSizeWhenWindowResized, updateSidebarSize]);
+
+    return <></>;
```

Comment:
nit can we return null instead of empty jsx?

Reply:
I changed it to return null.